app/MMB/src/scenes/timeline/Timeline.js

```diff
@@ -53,35 +54,53 @@ function getValidTimelineEvent(data) {
   return null;
 }
 
-function renderDaySeparator(event, prevEvent) {
-  if (event && event.timestamp && prevEvent && prevEvent.timestamp) {
-    const eventTimestamp = new Date(event.timestamp);
-    const prevEventTimestamp = new Date(prevEvent.timestamp);
-    if (!DateUtils.isSameDay(eventTimestamp, prevEventTimestamp)) {
-      return <DaySeparator date={eventTimestamp} />;
+function renderDaySeparator(renderedEvent, prevRenderedEvent) {
+  const timestamp = idx(renderedEvent, _ => _.props.data.timestamp);
+  if (!prevRenderedEvent) {
+    if (timestamp) {
+      return <DaySeparator date={new Date(timestamp)} />;
+    }
+  } else {
+    const prevTimestamp = idx(prevRenderedEvent, _ => _.props.data.timestamp);
+    if (timestamp && prevTimestamp) {
+      const eventTimestamp = new Date(timestamp);
+      const prevEventTimestamp = new Date(prevTimestamp);
+      if (!DateUtils.isSameDay(eventTimestamp, prevEventTimestamp)) {
+        return <DaySeparator date={eventTimestamp} />;
+      }
     }
   }
   return null;
 }
 
 function renderChildren(events) {
   if (events) {
-    return events.map((event, index) => {
-      if (event && event.timestamp && event.__typename) {
-        let daySeparator;
-        if (index === 0) {
-          daySeparator = <DaySeparator date={new Date(event.timestamp)} />;
-        } else {
-          daySeparator = renderDaySeparator(event, events[index - 1]);
-        }
-        return (
-          <React.Fragment key={'TimelineEvent-' + event.__typename + index}>
-            {daySeparator}
-            {getValidTimelineEvent(event)}
-          </React.Fragment>
+    const renderedEvents = events
+      .map(event => getValidTimelineEvent(event))
+      .filter(renderedEvent => renderedEvent);
```

Comment:
Why is this needed?

Reply:
There could be some additional events given by the GraphQL server which would not correspond to any implemented event component.
There could be situations where that adds an additional day banner when no component is being rendered, as `getValidTimelineEvent` would return `null`; also, since the first and last events need to display the vertical bars in a different way, the length of the non-null components must be taken into account, not the length of the events.
Does that make sense?

Reply:
Yes

Reply:
Maybe you could add a comment explaining it